pkgs/by-name/al/alp/package.nix

```diff
@@ -0,0 +1,40 @@
+{ stdenv
+, lib
+, buildGoModule
+, fetchFromGitHub
+, runCommand
+, alp
+}:
+buildGoModule rec {
+  pname = "alp";
+  version = "1.1.17";
+  src = fetchFromGitHub {
+    owner = "gernotfeichter";
+    repo = pname;
+    rev = version;
+    hash = "sha256-7lyWu1bVn7UwLb/Em6VBbg3FrMyxGjebxt5gJhm/xpI=";
+  };
+  vendorHash = "sha256-a2CQZKN/rPWh/Pn9gXfSArTCcGST472tsz1Kqm7M4vM=";
+  sourceRoot = "${src.name}/linux";
+  checkPhase = ''
+    go test -run Test_main_init
+  '';
+  buildPhase = ''
+    go build -o $GOPATH/bin/${pname} main.go
```

Comment:
The default buildPhase executes all go tests. The thing is that my go tests are integrative and require a docker daemon (test containers) currently, so I exempted the test execution in the buildPhase and instead I am running the remaining tests (non-docker-tests) in the checkPhase. This was the easiest solution to the problem I could find.

Notice that fetchFromGitHub uses only a tagged version. And since the tagging only happens after having successfully run goreleaser(required for distribution of my releases on non-nixos-systems; which I run currently on my machine with the docker tests enabled), it will not come to the situation where the (full) tests were not positive before running a release on nixpkgs.

Nevertheless I could implement improvements, e.g.:
- Maybe it is doable to run a docker daemon, just for the sake of the integration tests? So far I did not investigate the limits of those package tests.
- Maybe I could move those integration tests from being "tests" to being regular go code (binary).
- Or activate them only based on let's say some environment variable.
Do you have any preference/experience regarding that?

Reply:
Hum, indeed Docker is too huge.

Nonetheless, you should explain why you did this - not only to me, but in the form of a comment.

Something like 

```nix
#  Executing Go commands directly, because the default testsuite requires Docker
```

Reply:
I added
```
  # Executing Go commands directly in checkPhase and buildPhase below,
  # because the default testsuite runs all go tests, some of which require docker.
  # Docker is too expensive for https://github.com/NixOS/ofborg.
```